lib/browser-auto-tracking.js

```diff
@@ -87,6 +87,122 @@ export default function initAutoTrackingCore(lib) {
       cookie.set('initialReferrer', initialReferrer, cookieDomain);
     }
 
+    //todo (clarity): should all this code live somewhere else?
+    //todo (clarity): setup sampling
+    if (options.clarity_sample_rate > 0.0) {
+      const microsoft_project_id = "XXXXXXXXXX";  //not needed, example "f579s7kgqn"
+      //todo (clarity): how will this endpoint look in production?
+      const clarity_upload_endpoint = `https://clarity-initial-test.datapipe.prodperfect.com/clarity-initial-test-0/${session_uuid}`;
+      //todo (clarity): how are we going to manage/update this (store in our own S3?)
+      const clarity_tracking_code = "https://www.clarity.ms/s/0.7.8/clarity.js";
+
+      //todo (clarity): how do I make directly calling this code work?
+      const clarity_loading_script = `
+        !(function (c, l, a, r, i, t, y) {
+          if (a[c].v || a[c].t) return a[c]("event", c, "dup." + i.projectId);
+          (a[c].t = !0),
+              ((t = l.createElement(r)).async = !0),
+              (t.src = clarity_tracking_code),
+              (y = l.getElementsByTagName(r)[0]).parentNode.insertBefore(t, y), a[c]("start", i),
+              a[c].q.unshift(a[c].q.pop());
+        })("clarity", document, window, "script", {
+          projectId: microsoft_project_id,
+          upload: clarity_upload_endpoint,
+          expire: 365,
+          cookies: ["_uetmsclkid", "_uetvid"],
+          track: true,
+          lean: false,
+          content: true,
+          extract: [3, 1, ["4f509md79", "zjy56jaz", "4s4zm72hj"]],
+        });
+      `;
+
+      !(function(c,l,a,r,i,t,y){
+        c[a]=c[a]||function(){
+          (c[a].q=c[a].q||[]).push(arguments)};
+          t=l.createElement(r);
+          t.async=1;
+          t.text=clarity_loading_script;
+          y=l.getElementsByTagName(r)[0];
+          y.parentNode.insertBefore(t,y);
+      })(window, document, "clarity", "script", microsoft_project_id);
+
+      //todo (clarity): how should we actually get/set cookies? (this is taken from clarity)
+      function getCookie(key) {
+        let cookies = document.cookie.split(';');
+        if (cookies) {
```

Comment:
Not sure this check is really all that helpful in this scenario. If `document.cookie` is not defined, attempting to call split on it will blow things up. And if `document.cookie` is a string then I think this will always return an array which will never fail the check because `if ([])` evaluates to `true`.

Reply:
If you want to check that the array is not empty, you would need to do something like `if (cookies && cookies.length) { // do stuff };`

Reply:
FYI, `getCookie` and `setCookie` are copy-pasted directly from Clarity, seehttps://github.com/microsoft/clarity/blob/db16d1b8281530c96e577fe64fb8a785f6142f8f/packages/clarity-js/src/data/metadata.ts#L229